### PR TITLE
Feature/docsv1

### PR DESCRIPTION
**Pull request by amjadraza**
# GPT-ENGINEER : Documentations

The first stage of Docs for `get-engineer` project is included in this PR. The following have been done

## First Stage

1. Set up `sphinx` based project documentation
2. Generating the Package API docs using a Python Script
3. Introduction about the Package
4. Both `.rst` and `.md` files structures supported
5. Re-Factored the `pyproject.yml` to incorporate `docs` dependencies.

After Merge to do

- [ ] Generate and Host Project Documentation on [`readthedocs`](https://readthedocs.org/dashboard/) server from main branch
- [ ] Close relevant issue. 

## Second Stage:

- Re-Factor the Package code to include the `docstrings`
- Add more information and tutorials on using` gpt-engineer`

@raremojo thanks for your notes.

### CODE REVIEW

**Comment by amjadraza** [comment]
@AntonOsika please see, happy to discuss if needed.

File: docs/DOCS_BUILDING.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+Building Docs with Sphnix
```
Comment (QML-Coder) [inline docs/DOCS_BUILDING.md:1]: Typo: Sphnix -> Sphinx
Correct also at all other locations below.

Comment (amjadraza) [reply docs/DOCS_BUILDING.md]: Done

**Comment by amjadraza** [comment]
@AntonOsika Fixed the Links issues.

**Comment by AntonOsika** [comment]
Looks good with all the links etc!

I'm getting this error when running make build:
image

File: .github/FUNDING.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
 # These are supported funding model platforms
 
 github: [antonosika]
-patreon: gpt-engineer
+patreon: user?u=2497970
```
Comment (QML-Coder) [inline .github/FUNDING.yml:4]: Is this legit?

**Comment by amjadraza** [comment]
> Looks good with the ../ linking to md files etc!
> 
> I'm getting this error when running make build: image

@AntonOsika did you install the dependencies? Looks like the theme package is not installed correctly. Check if the package is installed correctly. 

I have created docs using the recent push. 

https://gpt-engineer.readthedocs.io/en/latest/index.html

If installing manually, use the `docs/requirements.txt` file for the doc dependencies. 

Thanks

**Comment by AntonOsika** [comment]
Great, it does work with requirements.txt

How about we only put the doc requirements in requirements.txt or in pyproject.toml, not both @amjadraza ?

**Comment by AntonOsika** [comment]
Hey @amjadraza .

I created a new venv and verified it does work!

Let's merge. 🎉 

About the requirements.txt, should we keep it do you think?

**Pull request merged**